### PR TITLE
Refactor block

### PR DESCRIPTION
**Pull request by lucafavatella**
Please refer to commit messages for details.

### CODE REVIEW

File: apps/ae_core/src/consensus/chain/block.erl [hunk 1]
```diff
@@ -107,6 +107,8 @@ lgh(1, X) ->
 lgh(N, X) ->
     lgh(N div 2, X+1).
 
+read_int_external(N) ->
+    (read_int(N))#block{trees = trees:new(0, 0, 0, 0, 0, 0)}.
```
Comment (michalzajda) [inline apps/ae_core/src/consensus/chain/block.erl]: are you sure this is correct? Can you elaborate

File: apps/ae_core/src/consensus/chain/block.erl [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 -export([block_to_header/1, test/0,
          height/1, prev_hash/1, txs/1, trees_hash/1, time/1, difficulty/1, comment/1, version/1, pow/1, trees/1, prev_hashes/1, 
-         read_int/2, read_int/1, hash/1, read/1, initialize_chain/0, make/4,
+         read_int/2, read_int/1, read_int_external/1, hash/1, read/1, initialize_chain/0, make/4,
```
Comment (michalzajda) [inline apps/ae_core/src/consensus/chain/block.erl]: we need to change those names, they say nothing. What is read int? and external variant

Comment (cytadela8) [inline apps/ae_core/src/consensus/chain/block.erl]: I would propose changing `read_int` to `get_by_height` and `read` to `get_by_hash`.

**Comment by lucafavatella** [comment]
I am splitting in separate more focused PRs. Thanks for your feedback.

**Pull request closed without merging**